lib/ansible/modules/cloud/amazon/ssm_send_command.py

```diff
@@ -0,0 +1,216 @@
+
+    invoke_params = {}
+
+    if document_name:
+        invoke_params['DocumentName'] = document_name
+    if comment:
+        invoke_params['Comment'] = comment
+    if instance_ids:
+        invoke_params['InstanceIds'] = instance_ids
+    if parameters:
+        invoke_params['Parameters'] = parameters
+
+    try:
+        response = client.send_command(**invoke_params)
+    except botocore.exceptions.ClientError as ce:
+        if ce.response['Error']['Code'] == 'ResourceNotFoundException':
+            module.fail_json(msg="Could not find the SSM doc to execute. Make sure "
+                             "the document name is correct and your profile has "
+                             "permissions to execute SSM.",
+                             exception=traceback.format_exc(ce))
+        module.fail_json(msg="Client-side error when invoking SSM, check inputs and specific error",
+                         exception=traceback.format_exc(ce))
+    except botocore.exceptions.ParamValidationError as ve:
+        module.fail_json(msg="Parameters to `invoke` failed to validate",
+                         exception=traceback.format_exc(ve))
+    except Exception as e:
+        module.fail_json(msg="Unexpected failure while invoking SSM send command.",
+                         exception=traceback.format_exc(e))
+
+    if await_return:
+        command_id = response['Command']['CommandId']
+        list_params = {}
+        if command_id:
+            list_params['CommandId'] = command_id
+            list_params['Details'] = True
+            checking = True
+            while checking:
+                try:
+                    invoke_response = client.list_command_invocations(**list_params)
+                except Exception as e:
+                    module.fail_json(msg="Error in checking on execution status",
+                                     exception=traceback.format_exc(e))
+                if invoke_response['CommandInvocations'][0]['Status'] == 'Success':
+                    checking = False
+                if invoke_response['CommandInvocations'][0]['Status'] == 'Failed':
+                    checking = False
+                    module.fail_json(msg="SSM Command failed", exception=invoke_response)
+                sleep(5)
+        else:
+            module.fail_json(msg='A valid command invocation ID was not returned.'
+                                 'Check the EC2 console command history')
+        results ={
+            'status': invoke_response['CommandInvocations'][0]['Status'],
+            'output': invoke_response['CommandInvocations'][0]['CommandPlugins'][0]['Output'],
+        }
+    else:
+        results ={
+            'status': response['Command']['Status'],
+            'output': '',
+        }
+
+    module.exit_json(changed=True, result=results)
+
+# import module snippets
+from ansible.module_utils.basic import *
```

Comment:
Need to use new style module imports

```
from ansible.module_utils.basic import AnsibleModule
from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
```

preferably at the top of the file to avoid flake8 complaints.